[llvm-5] build error on macOS

Build failure on macOS (10.11.6) and llvm-5.0rc2. I'm just using the gcc and g++ which came with my system, so I guess others will be in the same situation.

```sh
> git log -n 1
commit 5ca7c094ae75b6dab3647775a40cc6b8c00b0ab4
Merge: 4b3460b 5f58890
Author: Moritz Kiefer <moritz.kiefer@purelyfunctional.org>
Date:   Sun Aug 20 12:03:51 2017 +0200

    Merge pull request #129 from cocreature/speculatable

    Fix decoding of Speculatable attribute

> gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

> g++ --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 7.3.0 (clang-703.0.31)
Target: x86_64-apple-darwin15.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

> llc --version
LLVM (http://llvm.org/):
  LLVM version 5.0.0
  Optimized build with assertions.
  Default target: x86_64-apple-darwin15.6.0
  Host CPU: ivybridge

  Registered Targets:
    <all>

> stack build
llvm-hs-5.0.0: build (lib)
Log files have been written to: /Users/trevor/Dropbox/Projects/llvm/llvm-hs/.stack-work/logs/

--  While building package llvm-hs-5.0.0 using:
      /Users/trevor/Dropbox/Projects/llvm/llvm-hs/llvm-hs/.stack-work/dist/x86_64-osx/Cabal-1.24.2.0/setup/setup --builddir=.stack-work/dist/x86_64-osx/Cabal-1.24.2.0 build lib:llvm-hs --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/trevor/Dropbox/Projects/llvm/llvm-hs/.stack-work/logs/llvm-hs-5.0.0.log

    Preprocessing library llvm-hs-5.0.0...

    /Users/trevor/Dropbox/Projects/llvm/llvm-hs/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp:119:41: error:
         error: non-constant-expression cannot be narrowed from type 'int' to 'bool' in initializer list [-Wc++11-narrowing]
            CompileLayerT(Arg &&... arg) : data{std::forward<Arg>(arg)...} {}
                                                ^~~~~~~~~~~~~~~~~~~~~~

    /Users/trevor/Dropbox/Projects/llvm/llvm-hs/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp:192:16: error:
         note: in instantiation of function template specialization 'CompileLayerT<llvm::orc::CompileOnDemandLayer<CompileLayer, llvm::orc::JITCompileCallbackManager, llvm::orc::IndirectStubsManager> >::CompileLayerT<CompileLayer &, (lambda at src/LLVM/Internal/FFI/OrcJITC.cpp:194:9), llvm::orc::JITCompileCallbackManager &, std::__1::function<std::__1::unique_ptr<llvm::orc::IndirectStubsManager, std::__1::default_delete<llvm::orc::IndirectStubsManager> > ()> &, int &>' requested here
            return new CompileLayerT<LLVMCompileOnDemandLayer>(
                       ^

    /Users/trevor/Dropbox/Projects/llvm/llvm-hs/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp:119:41: error:
         note: insert an explicit cast to silence this issue
            CompileLayerT(Arg &&... arg) : data{std::forward<Arg>(arg)...} {}
                                                ^~~~~~~~~~~~~~~~~~~~~~
                                                static_cast<bool>(    )

    /Users/trevor/Dropbox/Projects/llvm/llvm-hs/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp:95:9: error:
         warning: ignoring return value of function declared with warn_unused_result attribute [-Wunused-result]
                data.emitAndFinalize(handles.lookup(h));
                ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~

    /Users/trevor/Dropbox/Projects/llvm/llvm-hs/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp:68:5: error:
         note: in instantiation of member function 'LinkingLayerT<llvm::orc::RTDyldObjectLinkingLayer>::emitAndFinalize' requested here
            LinkingLayerT(T data_) : data(std::move(data_)) {}
            ^

    /Users/trevor/Dropbox/Projects/llvm/llvm-hs/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp:262:16: error:
         note: in instantiation of member function 'LinkingLayerT<llvm::orc::RTDyldObjectLinkingLayer>::LinkingLayerT' requested here
            return new LinkingLayerT<RTDyldObjectLinkingLayer>(RTDyldObjectLinkingLayer(
                       ^
    1 warning and 1 error generated.
    `gcc' failed in phase `C Compiler'. (Exit code: 1)
```

--- a/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp
+++ b/llvm-hs/src/LLVM/Internal/FFI/OrcJITC.cpp
@@ -196,7 +196,8 @@ CompileLayer *LLVM_Hs_createCompileOnDemandLayer(
             partitioningFtor(&f, &result);
             return result;
         },
-        *callbackManager, *stubsManager, cloneStubsIntoPartitions);
+        *callbackManager, *stubsManager,
+        static_cast<bool>(cloneStubsIntoPartitions));
 }
 
 CompileLayer *LLVM_Hs_createIRTransformLayer(CompileLayer *compileLayer,

